packages/typespec-python/src/emitter.ts

```diff
@@ -608,7 +608,7 @@ function addLroInformation(
     const lroMeta = getLroMetadata(context.program, tspOperation);
     if (!isAzureCoreModel(lroMeta!.logicalResult)) {
         emittedOperation["responses"][0]["type"] = getType(context, lroMeta!.logicalResult);
-        if (lroMeta!.finalStep?.target.kind === "ModelProperty") {
+        if (lroMeta!.finalStep?.target.kind === "ModelProperty" && emittedOperation.method !== "PUT") {
```

Comment:
We have to add the logic until https://github.com/Azure/typespec-azure/pull/3370/files#r1287965512 is fixed.